### PR TITLE
add create_optimization_dataset.py

### PR DESCRIPTION
**Pull request by yudongqiu**
@dgasmith I'm not exactly sure if this is what you expect, but I added a script that converts the input molecules and cmiles_ids into arguments of the `client.add_procedure()` function, such as:
```
("optimization", "geometric", options, initial_molecule)
```
If there's an example to build the `OptimizationDataset`, I can try to convert this to a script that directly builds an `OptimizationDataset` from the JSON file, but I don't have a way to test it yet.

### CODE REVIEW

**Comment by dgasmith** [comment]
Can you migrate this into building a [OptimizationDataset](https://github.com/MolSSI/QCFractal/blob/master/qcfractal/tests/test_collections.py#L483)? Index is the canonical smiles index, is this right @ChayaSt?

**Comment by ChayaSt** [comment]
@dgasmith, for the index, do you mean the label of the job? 
Since these are unrestrained optimization, the labels should just be `cmiles_identifiers['canonical_isomeric_smiles']`.

**Comment by yudongqiu** [comment]
@ChayaSt Each molecule have one unique `cmiles_identifiers`, but may have multiple comformers. If we run one geometry optimization for each conformer, that means some of the some of them will have duplicate `cmiles_identifiers['canonical_isomeric_smiles']`. Is that fine?

**Comment by yudongqiu** [comment]
I guess we want to continue to use `{index}-1`, `{index}-2` to indicate conformers for the same molecule?

**Comment by ChayaSt** [comment]
Yes, that is a solution.

**Comment by yudongqiu** [comment]
@dgasmith I just rewrote the script. Since I'm not finally submitting it, could you please try that and see if it works?
Two things that might need to be updated:
1. The client configure file is not included here
2. The conda environment showing in requirements.txt might not be the same between @ChayaSt who runs `generate.py` and @dgasmith who will run `create_optimization_dataset.py`

**Comment by yudongqiu** [comment]
@dgasmith I just tested the creation script on my local server and fixed an issue. It should be good now.
Screen Shot 2019-06-28 at 11 27 35 PM

File: 6-25-19-smirnoff99Frost-coverage/create_optimization_dataset.py [hunk 1]
```diff
@@ -0,0 +1,102 @@
...
+    ds: qcportal.collections.OptimizationDataset
+        The dataset created
+
+    """
+    client = ptl.FractalClient.from_file(client_config_file)
+    # create a new dataset with specified name
+    print(f"Creating OptimizationDataset < {dataset_name} >")
+    ds = ptl.collections.OptimizationDataset(dataset_name, client=client)
+    # create specification for this dataset
+    opt_spec = {"program": "geometric"}
```
Comment (dgasmith) [inline 6-25-19-smirnoff99Frost-coverage/create_optimization_dataset.py]: Do we need to specify other geometric options here for this dataset?

File: 6-25-19-smirnoff99Frost-coverage/create_optimization_dataset.py [hunk 2]
```diff
@@ -0,0 +1,102 @@
...
+    qc_spec = {"driver": "gradient", "method": qm_method, "basis": qm_basis, "program": "psi4"}
```
Comment (dgasmith) [inline 6-25-19-smirnoff99Frost-coverage/create_optimization_dataset.py]: Do we need bond-orders or similar computations for this dataset?

File: .gitignore [hunk 3]
```diff
@@ -102,3 +102,6 @@ venv.bak/
 
 # mypy
 .mypy_cache/
+
+# vscode
+.vscode/
```
Comment (dgasmith) [inline .gitignore]: As a note, standard practice is to add this files to your own personal `~/.git/info/exclude`. I don't particularly mind, but you will get pushback in other OSS projects.

File: 6-25-19-smirnoff99Frost-coverage/create_optimization_dataset.py (hunk 2 above)
Comment (yudongqiu) [reply 6-25-19-smirnoff99Frost-coverage/create_optimization_dataset.py]: I think the bond orders are not used in the fitting, but if @ChayaSt want that information we can add that.

File: 6-25-19-smirnoff99Frost-coverage/create_optimization_dataset.py (hunk 1 above)
Comment (yudongqiu) [reply 6-25-19-smirnoff99Frost-coverage/create_optimization_dataset.py]: I think the default options should be good for this free optimization with no constraints.

File: .gitignore (hunk 3 above)
Comment (yudongqiu) [reply .gitignore]: I see. Thanks for the advice!

**Pull request merged**